### PR TITLE
Rubocop updates synced with MAIN

### PR DESCRIPTION
**Pull request by Ayushi1296**

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -1,51 +1,42 @@
...
 require:
   # - rubocop-capybara
-  # - rubocop-factory_bot
-  - rubocop-performance
+  # - rubocop-factory_bot  # requires RuboCop update
   - rubocop-rails
   # - rubocop-rake
-  - rubocop-rspec
-  # - rubocop-rspec_rails
+  # - rubocop-rspec
```
Comment (korny) [inline .rubocop.yml:7]: We can enable this now for everyone.

File: .rubocop.yml [hunk 2]
```diff
@@ -142,184 +108,125 @@ Naming/MethodParameterName:
...
     - development
-    - test
     - preview
-    - testing
-    - staging
     - production
-
+    - staging
+    - test
+    - testing
```
Comment (korny) [inline .rubocop.yml:144]: These are different in KIDS and MAIN 🤔

File: .rubocop.yml (hunk 1 above)
Comment (Ayushi1296) [reply .rubocop.yml:7]:
```suggestion
  - rubocop-rspec
```

File: .rubocop.yml (hunk 2 above)
Comment (Ayushi1296) [reply .rubocop.yml:144]: True, we can have it explicit in the rubocop yml on MAIN

Comment (korny) [reply .rubocop.yml:144]: preview and testing are not available in KIDS. But I guess we just have to allow all of them here.

Comment (Ayushi1296) [reply .rubocop.yml:144]: just for `testing` & `preview`

**Review by korny — APPROVED** [review]

File: .rubocop.yml (hunk 2 above)
Comment (korny) [reply .rubocop.yml:144]: Ah yes, there is a way to merge/extend inherited rules! Let me check.

Comment (korny) [reply .rubocop.yml]: Maybe like this? https://github.com/sofatutor/sofatutor/pull/14148

**Pull request merged**